app/src/main/java/org/wikipedia/main/MainFragment.kt

```diff
@@ -465,7 +476,26 @@ class MainFragment : Fragment(), BackPressedHandler, MenuProvider, FeedFragment.
     }
 
     override fun donateClick() {
-        (requireActivity() as? BaseActivity)?.launchDonateDialog()
+        if (showDonationBottomSheet) {
+            (requireActivity() as? BaseActivity)?.launchDonateDialog()
+        } else {
+            DonateDialog.launchDonateLink(requireContext())
+        }
+    }
+
+    private fun checkDonateConfig() {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
This seems like a lot of highly specific business logic being placed into `MainFragment`. Can we put this into a ViewModel, or into `DonationConfigHelper`?

Reply:
You are right. Moved to `DonateViewModel`.